Sources/Brave/Frontend/Browser/Tab.swift

```diff
@@ -384,45 +373,27 @@ class Tab: NSObject {
     if webView.backForwardList.responds(to: selector) {
       webView.backForwardList.performSelector(onMainThread: selector, with: nil, waitUntilDone: true)
     }
+    
+    // Remove the tab history from saved tabs
+    SessionTab.update(tabId: id, interactionState: webView.sessionData ?? Data(), title: title, url: webView.url ?? TabManager.ntpInteralURL)
   }
 
-  func restore(_ webView: WKWebView, restorationData: SavedTab?) {
+  func restore(_ webView: WKWebView, restorationData: (title: String, interactionState: Data)?) {
```

Comment:
I believe we can separate title and interactionState and not hold a tuple like I mentioned above.